product_pricelist_fixed_price/tests/test_product_pricelist_fixed_price.py

```diff
@@ -40,24 +41,27 @@ def test_onchange_base_ext(self):
         self.assertEqual(item.price_discount, -1.0)
 
         # change the type of the pricelist => purchase
-        plv.pricelist_id.type = 'purchase'
-        base_pur = item_obj._get_default_base({'type': 'purchase'})
-        self.assertNotEqual(base_sale, base_pur)
+        if module_obj.search_count([
```

Comment:
It's better for the diff to reverse the condition `if not...` and return early if so.

Reply:
You might also consider splitting the method in one testing sales prices and one testing purchase prices (following the principles of clean coding).  Then in the test for purchase prices do the early return as recommended by Pedro. 

If you want to clean up the code some more, you can extract a method that does the testing for a certain pricetype, because the code under the check... commented lines contain a lot of repetition.